plugins/hpi-cfu/fu-hpi-cfu-device.c

```diff
@@ -0,0 +1,1683 @@
+/*
+ * Copyright 2024 Pena Christian <christian.a.pena@hp.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "fu-cfu-struct.h"
+#include "fu-hpi-cfu-device.h"
+#include "fu-hpi-cfu-struct.h"
+
+#define GET_REPORT	   0x01
+#define SET_REPORT	   0x09
+#define FIRMWARE_REPORT_ID 0x20
+#define OFFER_REPORT_ID	   0x25
+#define END_POINT_ADDRESS  0x81
+
+#define IN_REPORT_TYPE	    0x0100
+#define OUT_REPORT_TYPE	    0x0200
+#define FEATURE_REPORT_TYPE 0x0300
+
+#define FU_HPI_CFU_PAYLOAD_LENGTH 52
+#define FU_HPI_CFU_DEVICE_TIMEOUT 0 /* ms */
+
+struct _FuHpiCfuDevice {
+	FuUsbDevice parent_instance;
+	gboolean claimed_interface;
+	guint8 iface_number;
+	FuHpiCfuState state;
+	guint8 force_version;
+	guint8 force_reset;
+	gint32 sequence_number;
+	gint32 currentaddress;
+	gint32 bytes_sent;
+	gint32 retry_attempts;
+	gint32 payload_file_size;
+	gint32 bytes_remaining;
+	gint32 last_packet_sent;
+	gint32 bulk_acksize;
+	gint32 curfilepos;
```

Comment:
there's a huge amount of type confusion here -- Coverity flags dozens of issues. All of these need to be the correct type, e.g. `gboolean` for TRUE/FALSE values, `gsize` for offsets or buffer lengths and `guint8` for single bytes. I started fixing these up but got bored after doing 10 or so.

Reply:
Could you please let us know how we run the coverity in our branch.

Reply:
I don't know how; I usually upload using https://scan.coverity.com/ -- if you fix the types then I can run the scan again and send you the results by email.